src/main/resources/api/data-repository-openapi.yaml

```diff
@@ -3815,6 +3920,48 @@ components:
       default: gcp
       description: Cloud platforms supported by TDR.
 
+    ##############################################################################
+    ## SEARCH API STANDARD MODELS
+    ##############################################################################
+    SearchIndexRequest:
+      required:
+        - sqlQuery
+      type: object
+      properties:
+        sqlQuery:
+          type: string
+          description: >
+            SQL to index BigQuery columns into Elasticsearch
+    SearchIndexModel:
+      type: object
+      properties:
+        indexSummary:
+          type: string
+          description: Summary of this index from Elasticsearch
+    SearchQueryRequest:
+      required:
+        - snapshotIds
+        - esQuery
+      type: object
+      properties:
+        snapshotIds:
+          minItems: 1
+          type: array
+          items:
+            $ref: '#/components/schemas/UniqueIdProperty'
+        esQuery:
+          type: string
+          description: >
+            Elasticsearch query to search against snapshot indices
+    SearchQueryResultModel:
+      type: object
+      properties:
+        esResult:
+          type: string
+          description: Elasticsearch results that match the filter criteria
```

Comment:
I would hide the fact that this is Elasticsearch. The user should't care

Reply:
Worth keeping in mind when we implement the endpoints.